src/main/java/seedu/address/storage/JsonAdaptedGood.java

```diff
@@ -52,7 +52,7 @@ public Good toModelType() throws IllegalValueException {
         }
         final GoodName modelGoodName = new GoodName(goodName);
 
-        if (!GoodQuantity.isValidGoodQuantity(goodQuantity)) {
+        if (!GoodQuantity.isValidGoodQuantity(String.valueOf(goodQuantity))) {
```

Comment:
So, do you want to keep consistently as string or int?

Reply:
I was thinking of keeping all constructor params as strings because it's on the strings that we do the majority of our invalid input checking. I think it would be harder to do validation on ints, and that is also inconsistent with most other classes that take Strings in the constructor.